src/java/src/main/java/com/dnanexus/DXSearch.java

```diff
@@ -900,11 +909,11 @@ public int size() {
         /**
          * Iterator implementation for findDataObjects results.
          */
-        private class ResultIterator
+        public class ResultIterator
```

Comment:
Please make this default-visibility and label it "@VisibleForTesting". That indicates it's being made available for tests and doesn't make it part of our public API.

Same goes for pageNo later.